pkgs/misc/vim-plugins/overrides.nix

```
@@ -334,6 +336,33 @@ self: super: {
     dependencies = with super; [ popfix ];
   });
 
+  # Usage:
+  # pkgs.vimPlugins.nvim-treesitter.withPlugins (with pkgs.tree-sitter.builtGrammars; [ tree-sitter-c ... ])
+  # or for all grammars:
+  # pkgs.vimPlugins.nvim-treesitter.withPlugins (builtins.attrValues pkgs.tree-sitter.builtGrammars)
+  nvim-treesitter = super.nvim-treesitter.overrideAttrs (old: {
+    passthru.withPlugins =
+      grammars:
+      self.nvim-treesitter.overrideAttrs (_: {
+        postPatch =
+          let
+            symLinks = map
```

```suggestion
      grammarFn:
      self.nvim-treesitter.overrideAttrs (_: {
        postPatch =
          let
            grammars = grammarFn super.tree-siter.builtGrammars;
            symLinks = map
```

I think having `withPlugins` take a function would make it more familiar and retain the current style of `withPlugins` from other packages (otherwise this lead to design dissonance I think)

---

@berbiche right, I took a look to `terraform.withPlugins` and that's the way is implemented. But I'm missing something here, with the changes you propose, you will call it like this:
```nix
pkgs.vimPlugins.nvim-treesitter.withPlugins (p: with pkgs.tree-sitter.builtGrammars; [ tree-sitter-c])
```
instead of
```nix
pkgs.vimPlugins.nvim-treesitter.withPlugins (with pkgs.tree-sitter.builtGrammars; [ tree-sitter-c ... ])
```

What is the advantage?

---

Actually, it would be used like this:
```nix
pkgs.vimPlugins.nvim-treesitter.withPlugins
  (p: with p; [ tree-sitter-c ]);
```

Just like for Python and possibly others

---

@berbiche got it, thanks for the example. PR updated